### DISCUSSION TITLE
Snow Inventory Agent Extension

### DISCUSSION THREAD

**Discussion by Tonrab** [mozilla/policy-templates#1131] [Q&A]
HI, I need to add the Snow Inventory agent to our estate, however, from reading thier notes, I can't see away to add it to the Intune ADMX template we have.  Am I missing something?
[https://docs.snowsoftware.com/release-notes/en/index-en.html#snowinventoryagentfirefoxextension1](url)

The first thing it mentions is to set saas.firefox.enabled to enable (under the **Install Browser extensions via Managment tool or Policy** header)
[https://docs.snowsoftware.com/snow-inventory-agents-and-oracle-scanners/en/UUID-788c7a9f-f5e5-1c1f-5562-aefd34798e39.html](url)
[https://docs.snowsoftware.com/documentation-resources/snow-inventory/configuration-doc.html](url)

and then amend the policies.json file with the following
`{
    "policies": {
        "3rdparty": {
            "Extensions": {
                cloudmetering@snowsoftware.com: {
                    "consentgiven": true
                }
            }
        },
        "ExtensionSettings": {
            cloudmetering@snowsoftware.com : {
                "install_url" : file://C:\\Program Files\\Snow Software\\Inventory\\Agent\\FFCloudmetering.xpi,
                "installation_mode" : "force_installed"
            }
        }
    }
}
`

Thanks for your help

**Comment by mkaply** [comment]
You can see how the ExtensionSettings stuff is done via Intune here:

https://mozilla.github.io/policy-templates/#extensionsettings

For their Extension specific changes, they should have created their own ADMX/ADML files to use.

**Comment by snowliver** [comment]
Hi @Tonrab,

join us in our community https://community.snowsoftware.com/s/question/0D5Td00000CLwnjKAD/we-are-looking-to-deploy-the-snow-inventory-agent-extension-for-firefox-users, and get noticed when we are updating our documentation and having ADMX/ADML files.

Cheers Oli

**Comment by mkaply** [comment]
@snowliver If y'all need any help with ADMX/ADML, let me know.

**Reply by snowliver** [reply]
Thanks @mkaply ,
I think, i have question. As you can see above, we use a bool value in storage.managed.
Is it possible to use ADMX to create something what will become a real boolean in FF`s storage managed? Seems ADMX windows registry only has int (DWORD) and string.

thanks in advance.

cheers Oli

**Comment by mkaply** [comment]
If you do something like this:

https://github.com/mozilla/policy-templates/blob/master/windows/firefox.admx#L4228

it does end up as a boolean in policy.

**Reply by snowliver** [reply]
Hi @mkaply  

i tried it like that and also took the example from https://extensionworkshop.com/documentation/enterprise/enterprise-development/#how-to-add-policy 

and it will be an integer.

![image](https://github.com/user-attachments/assets/dcbdab27-c148-4f36-b12f-d8a33526c827)

Also checked in the registry, how the built-in settings are created and they will be a DWORD set to 1. So i guess, we need to fetch that in the extension itself. :-(

Thanks for your help!

**Comment by mkaply** [comment]
Oh interesting. I didn't think about that side of things. Yeah, you're right. 

I think it's because on Chrome they use a schema and map it internally to the right type.

**Reply by snowliver** [reply]
Well, using a native storage manifest works, and using the policies.json works.

Do you know a way to use ADMX to satisfy the "a registry key with a single default value" wish  from "https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#windows" so someone could deploy the manifest directly?

I wasn't able to set the value on (Default) (or (Standard) on a german OS), only on creating new ValueNames. This is how it should look like:
![image](https://github.com/user-attachments/assets/9774f7b0-48fe-4282-b378-8f75bc97f586)

**Reply by mkaply** [reply]
I'm having a friend who knows more about this than me check into it.
